composer/callbacks/early_stopper.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+"""Early stopping callback."""
+
+from __future__ import annotations
+
+import logging
+from typing import Callable
+
+import numpy as np
+
+from composer.core import State
+from composer.core.callback import Callback
+from composer.loggers import Logger
+
+log = logging.getLogger(__name__)
+
+__all__ = ["EarlyStopper"]
+
+
+class EarlyStopper(Callback):
+    __doc__ = f"""Callback to halt training early.
```

Comment:
lets add a full docstring explaining usage and args.

also, is there a reason we're manually setting `__doc__` here?